2.0-proposal/copyright.md

```diff
@@ -0,0 +1,19 @@
+# Copyright in layered charms
+
+Each layer must state its copyright in a file named `LICENSE.layer-<layer-name>`. This is to ensure that `charm build` will preserve the copyright declaration of each layer.
+
+
+The built charm will contain all the licenses of every layer as shown in the following example.
```

Comment:
Whilst I really do like the idea of this, I do wonder if this opens up a massive can of worms.  Some licenses will be mutually (or otherwise) incompatible which means that the resulting 'charm' would essentially be (legally) in limbo - and couldn't be used.  I'm sure an actual lawyer would need to think about this.

I can't help feeling that all layers should be under a single license and contributors should just agree to use that license?  Choosing a license made sense for monolithic charms; I'm not sure it does for assembled charms, unless the layer license is akin to a library license?